lms/djangoapps/labster_course_license/views.py

```diff
@@ -81,15 +81,21 @@ def dashboard(request, course, ccx):
     return render_to_response('labster/course_license.html', context)
 
 
-def update_course(ccx, course_info, descriptors):
+def apply_field_overrides(ccx, course_info, descriptors):
     """
-    Updates all the descriptors.
+    Applies field overrides for the given descriptors.
     """
-    hidden = 'visible_to_staff_only'
+    field_name = 'visible_to_staff_only'
     for descriptor in descriptors:
         info = course_info[descriptor]
-        override_field_for_ccx(ccx, descriptor, hidden, info.is_hidden)
-        update_course(ccx, course_info, info.children)
+        if info.is_hidden:
+            override_field_for_ccx(ccx, descriptor, field_name, info.is_hidden)
+        else:
+            # Removes True values to get most actual ones from Master course.
```

Comment:
True -> False. And please write extended docstring (maybe in migrations) and link to the explanation from all places where do you change code.